src/index.js

```diff
@@ -248,12 +248,16 @@ export default class InputNumber extends React.Component {
   }
 
   onChange = (e) => {
+    const {
+      onChange,
+    } = this.props;
+
     if (this.state.focused) {
       this.inputting = true;
     }
-    const input = this.props.parser(this.getValueFromEvent(e));
-    this.setState({ inputValue: input });
-    this.props.onChange(this.toNumberWhenUserInput(input)); // valid number or invalid string
+    this.enteredInput = this.props.parser(this.getValueFromEvent(e));
```

Comment:
Should we be tracking `enteredInput` on state? Also not sure if this is the best variable name, perhaps `rawInput` is more descriptive?

Reply:
We need keep track of rawInput since it's used throughout the code.